source/backend/api/Services/DocumentSyncService.cs

```diff
@@ -94,113 +106,122 @@ public ExternalBatchResult SyncMayanMetadataTypes(SyncModel model)
 
-            ExternalResult<QueryResult<DocumentType>> mayanResult = await mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
+            ExternalBatchResult batchResult = new();
 
-            if (mayanResult.Status != ExternalResultStatus.Success && mayanResult.Payload?.Results?.Count == 0)
-            {
-                return new DocumentTypeSyncResponse();
-            }
+            Task<ExternalResult<QueryResult<DocumentType>>> documentTypeTask = mayanDocumentRepository.TryGetDocumentTypesAsync(pageSize: 5000);
+            documentTypeTask.Wait();
+
+            ExternalResult<QueryResult<DocumentType>> mayanDocumentTypes = documentTypeTask.Result;
 
             IList<PimsDocumentTyp> pimsDocumentTypes = documentTypeRepository.GetAll();
 
-            // Add the document types not in the backend
-            IList<PimsDocumentTyp> createdDocumentTypes = new List<PimsDocumentTyp>();
-            IList<PimsDocumentTyp> updatedDocumentTypes = new List<PimsDocumentTyp>();
-            foreach (var mayanDocumentType in mayanResult.Payload.Results)
+            // Add the document types not in mayan
+            IList<AddDocumentToMayanWithNameResponseModel> createResponseTasks = new List<AddDocumentToMayanWithNameResponseModel>();
+            IList<Task<ExternalResult<DocumentType>>> updateTasks = new List<Task<ExternalResult<DocumentType>>>();
+            foreach (var pimsDocumentTyp in pimsDocumentTypes)
             {
-                var matchingTypeFromModel = model.DocumentTypes.FirstOrDefault(x => x.Label.Equals(mayanDocumentType.Label, System.StringComparison.OrdinalIgnoreCase));
-                if (pimsDocumentTypes.FirstOrDefault(x => x.MayanId == mayanDocumentType.Id) == null)
+                var matchingTypeFromMayan = mayanDocumentTypes.Payload.Results.FirstOrDefault(x => x.Id == pimsDocumentTyp.MayanId);
+                if (matchingTypeFromMayan == null)
                 {
-                    var newPimsDocType = new PimsDocumentTyp() { MayanId = mayanDocumentType.Id, DocumentType = mayanDocumentType.Label, DisplayOrder = matchingTypeFromModel?.DisplayOrder };
-                    createdDocumentTypes.Add(documentTypeRepository.Add(newPimsDocType));
+                    createResponseTasks.Add(new AddDocumentToMayanWithNameResponseModel() { AddDocumentTypeTask = mayanDocumentRepository.TryCreateDocumentTypeAsync(new DocumentType() { Label = pimsDocumentTyp.DocumentTypeDescription }), Name = pimsDocumentTyp.DocumentType });
                 }
-                else if (pimsDocumentTypes.FirstOrDefault(x => x.MayanId == mayanDocumentType.Id && (mayanDocumentType.Label != x.DocumentType || x.DisplayOrder != matchingTypeFromModel?.DisplayOrder)) != null)
+                else if (matchingTypeFromMayan.Label != pimsDocumentTyp.DocumentTypeDescription)
                 {
                     // if the Mayan id is the same but the label or display order has changed, update the document type in PIMS.
```

Comment:
This is no longer true. If the Mayan id is the same but the label is different update the document type in mayan

Reply:
just the comment, the logic is sound